File: gencode/java/udmi/schema/DiscoveredNetworkFamily.java

package udmi.schema;

import javax.annotation.processing.Generated;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;


/**
 * Discovered Network Family
 * <p>
 * 
 * 
 */
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder({
    "addr",
    "scope"
})
@Generated("jsonschema2pojo")
public class DiscoveredNetworkFamily {

    @JsonProperty("addr")
    public String addr;
    @JsonProperty("scope")
    public String scope;

    @Override
    public int hashCode() {
        int result = 1;
        result = ((result* 31)+((this.addr == null)? 0 :this.addr.hashCode()));
        result = ((result* 31)+((this.scope == null)? 0 :this.scope.hashCode()));
        return result;
    }

    @Override
    public boolean equals(Object other) {
        if (other == this) {
            return true;
        }
        if ((other instanceof DiscoveredNetworkFamily) == false) {
            return false;
        }
        DiscoveredNetworkFamily rhs = ((DiscoveredNetworkFamily) other);
        return (((this.addr == rhs.addr)||((this.addr!= null)&&this.addr.equals(rhs.addr)))&&((this.scope == rhs.scope)||((this.scope!= null)&&this.scope.equals(rhs.scope))));
    }

}

File: gencode/python/udmi/schema/discovered_network_family.py
"""Generated class for discovered_network_family.json"""


class DiscoveredNetworkFamily:
  """Generated schema class"""

  def __init__(self):
    self.addr = None
    self.scope = None

  @staticmethod
  def from_dict(source):
    if not source:
      return None
    result = DiscoveredNetworkFamily()
    result.addr = source.get('addr')
    result.scope = source.get('scope')
    return result

  @staticmethod
  def map_from(source):
    if not source:
      return None
    result = {}
    for key in source:
      result[key] = DiscoveredNetworkFamily.from_dict(source[key])
    return result

  @staticmethod
  def expand_dict(input):
    result = {}
    for property in input:
      result[property] = input[property].to_dict() if input[property] else {}
    return result

  def to_dict(self):
    result = {}
    if self.addr:
      result['addr'] = self.addr # 5
    if self.scope:
      result['scope'] = self.scope # 5
    return result
